src/main/java/cz/cvut/kbss/termit/service/repository/UserRepositoryService.java

```diff
@@ -67,6 +69,10 @@ public boolean exists(String username) {
         return userAccountDao.exists(username);
     }
 
+    public Optional<UserAccount> findByUsername(String username) {
```

Comment:
This should call postLoad to ensure consistent behavior with other find methods.